src/cargo/core/compiler/job_queue.rs

```diff
@@ -764,6 +797,71 @@ impl<'cfg> DrainState<'cfg> {
                 // It doesn't really matter if this fails.
                 drop(cx.bcx.config.shell().status("Finished", message));
             }
+
+            if !self.future_incompat_reports.is_empty() {
```

Comment:
I assume you mean the `-Z` flag? We want to run this logic regardless of whether or not the user wants the report displayed, so that we can save the report to disk and print a notice.

Reply:
I mean this should be in the `if` block above.